lib/SILOptimizer/Transforms/SILMem2Reg.cpp

```diff
@@ -2208,6 +2125,14 @@ bool MemoryToRegisters::promoteAllocation(AllocStackInst *alloc,
   // Remove write-only AllocStacks.
   if (isWriteOnlyAllocation(alloc) && !lexicalLifetimeEnsured(alloc)) {
     LLVM_DEBUG(llvm::dbgs() << "*** Deleting store-only AllocStack: "<< *alloc);
+    if (alloc->getElementType().getASTType()->isUninhabited()) {
```

Comment:
Can you add a comment explaining this condition? (particularly the isUninhabited part?)